src/main/java/org/mule/module/json/internal/JsonSchemaValidatorJavaJsonToolsWrapper.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.json.internal;
+
+import static org.mule.module.json.api.JsonError.INVALID_SCHEMA;
+import static org.mule.module.json.api.JsonSchemaDereferencingMode.CANONICAL;
+import static org.mule.module.json.internal.ValidatorCommonUtils.resolveLocationIfNecessary;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static java.util.stream.Collectors.joining;
+import static com.fasterxml.jackson.databind.SerializationFeature.INDENT_OUTPUT;
+import static com.google.common.collect.Lists.newArrayList;
+
+import org.mule.module.json.api.JsonSchemaDereferencingMode;
+import org.mule.module.json.internal.error.SchemaValidationException;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.extension.api.exception.ModuleException;
+import java.io.InputStream;
+import java.util.Map;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonschema.core.exceptions.ProcessingException;
+import com.github.fge.jsonschema.core.load.Dereferencing;
+import com.github.fge.jsonschema.core.load.configuration.LoadingConfiguration;
+import com.github.fge.jsonschema.core.load.configuration.LoadingConfigurationBuilder;
+import com.github.fge.jsonschema.core.load.uri.URITranslatorConfiguration;
+import com.github.fge.jsonschema.core.load.uri.URITranslatorConfigurationBuilder;
+import com.github.fge.jsonschema.core.report.ProcessingReport;
+import com.github.fge.jsonschema.main.JsonSchema;
+import com.github.fge.jsonschema.main.JsonSchemaFactory;
+
+/**
+ * This Wrapper make validations with library java.json.tools
+ * Supports Json schema validations with version Draft V3 and V4
+ */
+public class JsonSchemaValidatorJavaJsonToolsWrapper extends JsonSchemaValidator {
+
+  private final JsonSchema jsonSchema;
+  private final ObjectMapper objectMapper;
+
+  public JsonSchemaValidatorJavaJsonToolsWrapper(ValidatorKey key, JsonNode jsonSchemaNode) {
+    super(key);
+    jsonSchema =
+        loadSchemaLibrary(jsonSchemaNode, super.getSchemaLocation(), super.getSchemaRedirects(), super.getDereferencing());
+    objectMapper = new ObjectMapper();
+  }
+
+  @Override
+  public void validate(InputStream inputStream) {
+    JsonNode jsonNode = super.asJsonNode(inputStream);
+    ProcessingReport report;
+    try {
+      report = jsonSchema.validate(jsonNode);
+
+    } catch (ProcessingException processingException) {
+      //TODO We must create a new error: INVALID_REFERENCE, to inform the user that the external references declared in the Schema cannot be accessed(W-12301483)
```

Comment:
Is there any reason not to complete the TODO before merging? @GuidoGianfrate

Reply:
Yes, because this TODO is for the next MAJOR